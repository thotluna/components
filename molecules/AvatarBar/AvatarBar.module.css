.avatarBar {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: flex-start;
  padding: 1rem;
  border: 2px solid var(--color-primary);
  border-radius: var(--border-radius-sm);
  gap: 2rem;
}
