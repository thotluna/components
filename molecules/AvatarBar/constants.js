export const options = {
  sizes: ['sm', 'lg'],
}
