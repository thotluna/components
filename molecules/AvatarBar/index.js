export { default } from './AvatarBar'
export { options } from './constants'
export { default as styles } from './AvatarBar.module.css'
