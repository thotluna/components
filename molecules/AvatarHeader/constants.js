export const options = {
  size: ['sm', 'md'],
}
