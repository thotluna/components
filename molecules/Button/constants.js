export const options = {
  colors: ['base', 'primary', 'secondary'],
}
