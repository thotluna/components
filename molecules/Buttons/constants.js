export const options = {
  size: ['full', 'short', 'mini'],
  color: ['primary', 'secondary', 'inverted'],
}
