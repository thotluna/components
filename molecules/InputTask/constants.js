export const status = {
  STATUS_STANDBY: 'standby',
  STATUS_CREATING: 'creating',
}

export const options = {
  status: [status.STATUS_STANDBY, status.STATUS_CREATING],
}
