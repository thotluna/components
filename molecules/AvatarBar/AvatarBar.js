import React from 'react'
import PropTypes from 'prop-types'
import styles from './AvatarBar.module.css'
import { options } from './constants'
import { getClasses } from '../../helpers/styles'

import Avatar from '../../atoms/Avatar'
import Title from '../../atoms/Title'
import Paragraph from '../../atoms/Paragraph'

export const AvatarBar = ({ user, size }) => {
  const getStyles = getClasses(styles)({ size })
  return (
    <div className={getStyles('avatarBar', ['size'])}>
      <Avatar src={user.avatar} size={size} />
      <div className={getStyles('avatar-data')}>
        <Title size={size}>{user.username}</Title>
        {size === 'lg' && (
          <Paragraph size="sm" color="inverted">
            {user.email}
          </Paragraph>
        )}
      </div>
    </div>
  )
}

AvatarBar.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
    email: PropTypes.string,
  }).isRequired,
  size: PropTypes.oneOf(options.sizes),
}

AvatarBar.defaultProps = {
  size: 'lg',
}

export default AvatarBar
