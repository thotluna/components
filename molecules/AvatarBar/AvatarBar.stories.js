import React from 'react'
import AvatarBar from './AvatarBar'
import { options } from './constants'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <AvatarBar {...args} />
const ListTemplate = getVariusTemplate(AvatarBar)

export default {
  title: 'Molecules/AvatarBar',
  avatarBar: AvatarBar,
  args: {
    user: {
      username: 'Eladio Feijoo',
      email: 'eladio.feijoo@gmail.com',
      avatar: 'https://i.pravatar.cc/300',
    },
    size: 'lg',
  },
  argTypes: {
    size: getOptionsArgTypes(options.sizes),
  },
}

export const Default = Template.bind({})

export const Size = ListTemplate.bind({})
Size.args = { items: options.sizes.map((size) => ({ size })) }
