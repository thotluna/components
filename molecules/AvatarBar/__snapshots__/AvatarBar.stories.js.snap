// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/AvatarBar Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="avatarBar size-lg"
  >
    <div
      className="avatar size-lg"
    >
      <img
        alt="your avatar"
        className="avatar size-lg"
        src="https://i.pravatar.cc/300"
      />
    </div>
    <div
      className="avatar-data"
    >
      <h2
        className="title color-base"
      >
        Eladio Feijoo
      </h2>
      <p
        className="paragraph color-inverted size-sm"
      >
        eladio.feijoo@gmail.com
      </p>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/AvatarBar Size 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="avatarBar size-sm"
  >
    <div
      className="avatar size-sm"
    >
      <img
        alt="your avatar"
        className="avatar size-sm"
        src="https://i.pravatar.cc/300"
      />
    </div>
    <div
      className="avatar-data"
    >
      <h4
        className="title color-base"
      >
        Eladio Feijoo
      </h4>
    </div>
  </div>
  <div
    className="avatarBar size-lg"
  >
    <div
      className="avatar size-lg"
    >
      <img
        alt="your avatar"
        className="avatar size-lg"
        src="https://i.pravatar.cc/300"
      />
    </div>
    <div
      className="avatar-data"
    >
      <h2
        className="title color-base"
      >
        Eladio Feijoo
      </h2>
      <p
        className="paragraph color-inverted size-sm"
      >
        eladio.feijoo@gmail.com
      </p>
    </div>
  </div>
</div>
`;
